### DISCUSSION TITLE
Aborted connection to db: '...' user: '...' host: '...' (Got an error reading communication packets)

### DISCUSSION THREAD

**Discussion by jowin202** [aio-libs/aiomysql#990] [Q&A] [answered]
Hello,
I try to connect my fastapi to mysql database with aiomysql.

The minimal code looks like

```
import aiomysql
import asyncio

async def connect_to_db():
    return await aiomysql.connect(host="localhost",port=3306,user="fastapi_user",password="fastapi_password", db="fastapi_db")

async def execute_query_and_fetch(query, params=None):
    conn = await connect_to_db()
    async with conn.cursor(aiomysql.DictCursor) as cursor:
        await cursor.execute(query, params)
        result = await cursor.fetchall()
        await cursor.close()
    conn.close()
    return result


print(asyncio.run(execute_query_and_fetch("SELECT * from users")))

```

This code is basically the same as this one here: https://aiomysql.readthedocs.io/en/stable/connection.html
whenever I call execute_query_and_fetch("SELECT ..."), I receive following error

> mysql-1       | 2024-08-18 12:39:22 9 [Warning] Aborted connection 9 to db: 'fastapi_db' user: 'fastapi_user' host: '172.19.0.4' (Got an error reading communication packets)

However, I get the correct result, but anyway. There shouldn't be an error message for each query.

Has anyone an idea what happened here?

**UPDATE: it works if I switch mariadb docker container to mysql, but anyway, is there a simple explanation?**

**Comment by Cycloctane** [comment, marked as answer]
aiomysql simply closes the tcp connection without sending [COM_QUIT](https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_com_quit.html) packet when `conn.close()` is called.
You can use `await conn.ensure_closed()` to avoid error messages. `ensure_closed()` can send COM_QUIT and then close the connection safely.
